sorc/ncep_post.fd/SURFCE.f

```diff
@@ -720,6 +721,32 @@ SUBROUTINE SURFCE
          endif
       ENDIF
 !
+!     TOTAL SOIL MOISTURE
+      IF (IGET(713)>0) THEN
+!$omp parallel do private(i,j)
+         DO J=JSTA,JEND
```

Comment:
@EricJames-NOAA Could you adapt the settings of UPP ID 036 (soil moisture) as:
```
 DO J=JSTA,JEND
           DO I=ISTA,IEND
             IF(SMSTOT(I,J)/=SPVAL) THEN
               IF(SM(I,J) > SMALL .AND. SICE(I,J) < SMALL) THEN
                 GRID1(I,J) = 1000.0  ! TEMPORY FIX TO MAKE SURE SMSTOT=1 FOR WATER
               ELSE
                 GRID1(I,J) = SMSTOT(I,J)
               END IF
             ELSE
               GRID1(I,J) = 1000.0
             ENDIF
           ENDDO
         ENDDO
```

Reply:
Shouldn't we preserve ID 036 for output of SOILMOI in kg/m3 in case anyone needs that?

Reply:
@EricJames-NOAA Yes, We can keep ID 036 to output SOILMOI for non-UFS based models. After your PR is merged for RRFS, I will update total soil moisture grib2 name for rest of UFS based applications.

Reply:
@WenMeng-NOAA ok.  Sorry if I don't understand...what changes should I make in this PR?

Reply:
@EricJames-NOAA Could you add the same settings of ID 036 in this new ID block? such as:
```
DO J=JSTA,JEND
           DO I=ISTA,IEND
             IF(SMSTOT(I,J)/=SPVAL) THEN
               IF(SM(I,J) > SMALL .AND. SICE(I,J) < SMALL) THEN
                 GRID1(I,J) = 1000.0  ! TEMPORY FIX TO MAKE SURE SMSTOT=1 FOR WATER
               ELSE
                 GRID1(I,J) = SMSTOT(I,J)
               END IF
             ELSE
               GRID1(I,J) = 1000.0
             ENDIF
           ENDDO
         ENDDO
```

Reply:
@WenMeng-NOAA I'm not sure we need all those checks if the SMSTOT is really in kg/m2.  In that case, the current code for ID 713 retains SPVAL over all water points, and only has values over land.  I don't think it makes sense to assign column integrated soil moisture to a numeric value over water, I think we should just leave them as missing values.

If SMSTOT is in kg/m3, then other applications can use the existing ID 036.

Reply:
@EricJames-NOAA Okay, let's keep RRFS as your current setting. The "GRID1(I,J) = 1000.0" is a legacy setting for GFS products. I will evaluate GFS products, coordinate with users and submit separate PR for gfs modification only if needed later. Please let me know if this approach works for you.

Reply:
@WenMeng-NOAA OK, thank you, that sounds like a good plan to me.